src/OpenLoco/Windows/TextInputWindow.cpp

```diff
@@ -243,6 +243,17 @@ namespace OpenLoco::Ui::Windows::TextInput
         position = { inputSession.xOffset, 1 };
         Gfx::drawStringLeft(*clipped, &position, Colour::black, StringIds::black_stringid, _commonFormatArgs);
 
+        position = { window.x + widget->left, window.y + widget->top - 10 };
+        drawnBuffer = (char*)StringManager::getString(StringIds::buffer_2039);
```

Comment:
This is pretty old code you have copied I would just do it as follows:
```cpp
auto args = FormatArguments();
args.push<uint16_t>(numCharacters);
args.push<uint16_t>(maxNumCharacters)
Gfx::drawStringLeft(*rt, &position, Colour::black, StringIds::num_characters_left_int_int, &args);
// Where StringIds::num_characters_left_int_int is "{COLOUR WINDOW_2}{UINT16}/{UINT16}"
```
This is much more flexible and doesn't involve messing about copying pointers of data and such.